scripts/pre_commit_linter.py

```diff
@@ -2075,93 +2135,36 @@ def _check_codeowner_file(self):
+                for file_path in file_paths:
+                    match = False
+                    for file_pattern in file_patterns:
+                        if file_path in glob.glob(file_pattern):
                             match = True
-                            continue
-                        for path_to_match in path_patterns:
-                            # The level of the file in the directory
-                            # structure. For e.g. /core/controllers/
-                            # domain.py is on third level.
-                            # This condition checks if the path to check
-                            # is a directory or a file. If it is a
-                            # file, the level would be the same as found
-                            # by calculating len(path_to_match.split('/'))
-                            # but is reduced by one if it is a directory
-                            # since the split command will return an empty
-                            # string at the last of the list which will
-                            # wrongfully increase the level.
-                            if path_to_match.split('/')[-1]:
-                                level = len(path_to_match.split('/'))
-                            else:
-                                level = len(path_to_match.split('/')) - 1
-                            # This condition finally matches the file being
-                            # walked currently against the path from the
-                            # CODEOWNERS file. The level helps in matching
-                            # by considering the file name upto only the
-                            # the level of the CODEOWNERS path. For e.g.
-                            # if the file being walked upon is /core/domain
-                            # /domain.py and the path from CODEOWNERS to be
-                            # matched is /core/ then it will only consider
-                            # the file name upto level 1 i.e. just the '/core'
-                            # part of the file name since it is sufficient to
-                            # be matched.
-                            if os.path.join(*((
-                                    os.path.join(root, file_name).replace(
-                                        './', '', 1)).split(
-                                            '/')[:level])) in glob.glob(
-                                                path_to_match):
-                                match = True
-                                break
-                        if not match and self.verbose_mode_enabled:
-                            print ('WARNING! %s/%s is not covered under '
-                                   'CODEOWNERS' % (root, file_name))
+                            break
+                    if not match and self.verbose_mode_enabled:
+                        print '%s is not covered under CODEOWNERS' % file_path
```

Comment:
I tried deleting a codeowner rule and ran the linter, it didn't catch it and passed the tests as usual. Is this expected behaviour?

Reply:
It depends on which rule you have deleted, assume someone is looking for oppia/storage and you have deleted oppia/storage/exploration, in this case, the test will not fail. Can you please tell which line you have deleted?

Reply:
I have deleted [L#170](https://github.com/oppia/oppia/pull/6616/files#diff-5cfb934ca0ec4c1743905a51dc428b84R170). And the checks pass. Let me know if I am missing something here.

Reply:
I tried the same and got this:
![image](https://user-images.githubusercontent.com/16653571/56899221-8cc4b200-6ab0-11e9-942c-54cd3d0809d7.png)

Can you please add the steps you are following?

Reply:
Ah never mind, the checks failed as expected. I pulled this PR's latest state and that way it worked!
Thanks!